addons/website/static/src/client_actions/website_preview/website_preview.js

```diff
@@ -40,24 +39,15 @@ export class WebsitePreview extends Component {
         useBus(this.websiteService.bus, 'UNBLOCK', () => this.unblock());
 
         onWillStart(async () => {
-            const [backendWebsiteRepr] = await Promise.all([
-                this.orm.call('website', 'get_current_website'),
-                this.websiteService.fetchWebsites(),
-                this.websiteService.fetchUserGroups(),
-            ]);
-            this.backendWebsiteId = unslugHtmlDataObject(backendWebsiteRepr).id;
-
-            const encodedPath = encodeURIComponent(this.path);
-            if (this.websiteDomain && this.websiteDomain !== window.location.origin) {
-                window.location.href = `${this.websiteDomain}/web#action=website.website_preview&path=${encodedPath}&website_id=${this.websiteId}`;
-            } else {
-                this.initialUrl = `/website/force/${this.websiteId}?path=${encodedPath}`;
+            await this.websiteService.fetchUserGroups();
+            if (this.websiteService.hasMultiWebsites) {
```

Comment:
I checked and it seems that Javascript has a restriction on the getter, they cannot be asynchronous.
I could not find why, maybe because it would add complexity, to cache what was fetched ?